draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -894,38 +893,32 @@ Elligator2 can be simplified with projective coordinates.
 
 ((TODO: write this variant))
 
-# Random Oracles
-
-## Interface
+# Random Oracles {#ffstv}
 
-The generic interface for deterministic encoding functions to elliptic curves is as follows:
+Some applications require a Random Oracle (RO) of points, which can be constructed
+from deterministic encoding functions. Farashahi et al. {{FFSTV13}} showed a
+generic mapping construction that is indistinguishable from a random oracle.
+In particular, let `f : {0,1}^* -> E(F)` be a deterministic encoding function,
+and let `H0` and `H1` be two hash functions modeled as random oracles that map
+bit strings to elements in the field `F`, i.e., `H0,H1 : {0,1}* -> F`. Then,
+the `hash2curveRO(alpha)` mapping is defined as
 
 ~~~
-hash2curve(alpha)
+hash2curveRO(alpha) = f(H0(alpha)) + f(H1(alpha))
 ~~~
 
-where alpha is a message to encode on a curve.
+where alpha is an octet string to be encoded as a point on a curve.
 
-## General Construction (FFSTV13) {#ffstv}
+## Interface
 
-When applications need a Random Oracle (RO), they can be constructed from deterministic encoding
-functions. In particular, let F : {0,1}^* -> E be a deterministic encoding function onto
-curve E, and let H0 and H1 be two hash functions modeled as random oracles that map input
-messages to the base field of E, i.e., Z_q. Farashahi et al. {{FFSTV13}} showed that the
-following mapping is indistinguishable from a RO:
+Using the deterministic encodings from {{encodings}}, the `hash2curveRO(alpha)`
+mapping can be instantiated as
 
 ~~~
-hash2curve(alpha) = F(H0(alpha)) + F(H1(alpha))
+hash2curveRO(alpha) = hash2curve(alpha || 0x02) + hash2curve(alpha || 0x03)
```

Comment:
Sense check: does this properly separate domains?
For example, using I can call regular Icart on inputs `alpha || 0x02`, `alpha || 0x03`, and add the outputs.

I think this would only be an issue in a situation where both were exposed in the same protocol?

Reply:
Agree, it is required to separate the two cases when using an encoding X with and without the random oracle property. 
Requires further investigation.